src/Microsoft.ML.Data/DataView/ArrayDataViewBuilder.cs

```diff
@@ -78,13 +78,27 @@ public void AddColumn<T>(string name, PrimitiveType type, params T[] values)
         public void AddColumn(string name, ValueGetter<VBuffer<ReadOnlyMemory<char>>> getKeyValues, ulong keyMin, int keyCount, params uint[] values)
         {
             _host.CheckValue(getKeyValues, nameof(getKeyValues));
-            _host.CheckParam(keyCount > 0, nameof(keyCount));
+            //_host.CheckParam(keyCount > 0, nameof(keyCount));
             CheckLength(name, values);
             _columns.Add(new AssignmentColumn<uint>(new KeyType(DataKind.U4, keyMin, keyCount), values));
             _getKeyValues.Add(name, getKeyValues);
             _names.Add(name);
         }
 
+        /// <summary>
+        /// Constructs a new key column from an array where values are copied to output simply
+        /// by being assigned.
+        /// </summary>
+        public void AddColumn(string name, ValueGetter<VBuffer<ReadOnlyMemory<char>>> getKeyValues, ulong keyMin, int keyCount, params ulong[] values)
```

Comment:
>AddColumn [](start = 20, length = 9)

You spend quite a lot of time to figure out what each parameter mean.
Don't you think it would be nice to provide description for each of them?

Reply:
Yes, especially since this is a public api.


---
In reply to: [243448690](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243448690) [](ancestors = 243448690)